CMakeLists.txt

```diff
@@ -1,24 +1,19 @@
-cmake_minimum_required(VERSION 3.8)
-project(bls12-381 VERSION 1.0.0)
+cmake_minimum_required(VERSION 3.12)
 
-#set(CMAKE_CXX_STANDARD 17)
-#set(CMAKE_CXX_EXTENSIONS ON)
-#set(CMAKE_CXX_STANDARD_REQUIRED ON)
-#add_compile_options(-Wall)
+if(CMAKE_SOURCE_DIR STREQUAL CMAKE_CURRENT_SOURCE_DIR AND NOT CMAKE_BUILD_TYPE AND NOT CMAKE_CONFIGURATION_TYPES)
+  set(CMAKE_BUILD_TYPE Release CACHE STRING "Build type" FORCE)
+  set_property(CACHE CMAKE_BUILD_TYPE PROPERTY STRINGS "Debug" "Release" "MinSizeRel" "RelWithDebInfo")
+endif()
 
-# compile with "-march" or enable cpu features explicitly for best performance:
-#set(CMAKE_CXX_FLAGS "-madx -mbmi2")
-set(CMAKE_CXX_FLAGS "-fstack-protector-strong")
+project(bls12-381 VERSION 1.0.0)
 
-if(NOT CMAKE_BUILD_TYPE)
-  set(CMAKE_BUILD_TYPE "Release"
-    CACHE STRING "Possible values are empty, Debug, Release, RelWithDebInfo, MinSizeRel, ..."
-    FORCE
-  )
-endif()
+file(GLOB BLS12-381_SOURCES CONFIGURE_DEPENDS src/*.cpp)
+file(GLOB BLS12-381_HEADERS CONFIGURE_DEPENDS include/bls12-381/*.hpp)
 
-# the library
-add_subdirectory(src)
+add_library(bls12-381 ${BLS12-381_SOURCES})
+target_include_directories(bls12-381 PUBLIC include)
+set_target_properties(bls12-381 PROPERTIES PUBLIC_HEADER "${BLS12-381_HEADERS}")
```

Comment:
Because of leap's obtuse cmake install rules due to libtester, I've elected not to do the `install()` here. Enough is exposed via this `PUBLIC_HEADER` property to, in the cmake project that is including `bls12-381`, simply do
```cmake
install(TARGETS bls12-381 ARCHIVE       DESTINATION "${CMAKE_INSTALL_FULL_LIBDIR}"
                          PUBLIC_HEADER DESTINATION "${CMAKE_INSTALL_FULL_INCLUDEDIR}/bls12-381")
```
This isn't perfect: using a `FILE_SET` would be better, but that requires newer cmake than our target.